proxy/src/cancellation.rs

```diff
@@ -107,6 +129,14 @@ impl<P: CancellationPublisher> CancellationHandler<P> {
             }
             return Ok(());
         };
+
+        // Seems cheaper not to spawn task sicne we don't have a lot of IPs in the allowlist (10)
```

Comment:
Spawn a task for what? Would we need to await something?

Reply:
Left a comment here for initial issue scope